src/services/spot/Spot.cpp

```diff
@@ -0,0 +1,387 @@
+//
+//  * Redistributions of source code must retain the above copyright notice, this list of
+//    conditions and the disclaimer below.
+//  * Redistributions in binary form must reproduce the above copyright notice, this list of
+//    conditions and the disclaimer (as noted below) in the documentation and/or other materials
+//    provided with the distribution.
+//  * Neither the name of the LLNS/LLNL nor the names of its contributors may be used to endorse
+//    or promote products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS
+// OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// LAWRENCE LIVERMORE NATIONAL SECURITY, LLC, THE U.S. DEPARTMENT OF ENERGY OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+// (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+// ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+// Spot.cpp
+// Outputs Caliper performance results to Spot-parseable formats
+
+
+#include "caliper/CaliperService.h"
+
+#include "caliper/Caliper.h"
+#include "caliper/SnapshotRecord.h"
+
+#include "caliper/reader/Aggregator.h"
+#include "caliper/reader/RecordSelector.h"
+#include "caliper/reader/CalQLParser.h"
+#include "caliper/reader/QueryProcessor.h"
+
+#include "caliper/common/Log.h"
+#include "caliper/common/Node.h"
+#include "caliper/common/OutputStream.h"
+#include "caliper/common/RuntimeConfig.h"
+#include "caliper/common/util/split.hpp"
+
+#include "rapidjson/rapidjson.h"
+#include "rapidjson/document.h"
+#include "rapidjson/ostreamwrapper.h"
+#include "rapidjson/writer.h"
+
+#include <iostream>
+#include <sstream>
+#include <iterator>
+#include <fstream>
+#include <time.h>
+using namespace cali;
+
+namespace
+{
+
+    class Spot {
+        static std::unique_ptr<Spot> s_instance;
+        static const ConfigSet::Entry  s_configdata[];
+        static int divisor;
+        template<typename T>
+        using List = std::vector<T>;
+        using AggregationHandler = cali::Aggregator*;
+        using SelectionHandler = cali::RecordSelector*;
+        using TimeType = unsigned int;
+        using SingleJsonEntryType = List<std::pair<std::string,TimeType>>;
+        using JsonListType = List<SingleJsonEntryType>;
+        using AggregationDescriptor = std::pair<std::string, std::string>;
+        using AggregationDescriptorList = List<AggregationDescriptor>;
+        static List<std::pair<AggregationHandler,SelectionHandler>> m_queries;
```

Comment:
Would be better to move all of the static things here into the spot object. That would make the multi-config conversion easier, where we will have to do that anyway.